Update the developer setup notes (#48)
badboy/glean.rs: docs/dev/android/setup-android-build-environment.md

> <!-- MarkdownTOC autolink="true" -->

Reviewer: Is this auto-generated by your local tool? 
It will [be rendered into a webpage](https://badboy.github.io/glean.rs/book/dev/android/setup-android-build-environment.html) and therefore the heading should be first. 
I'm also not sure we need an extensive inline-ToC for a now rather short doc.

Author: Oh, ok, removing this.

  - [Configure Maven](#configure-maven)
  - [FAQ](#faq)

<!-- /MarkdownTOC -->

# Setup the Android Build Environment

## Doing a local build of the Android Components:

This document describes how to make local builds of the Android components in
this repository. Most consumers of these components *do not* need to follow
this process, but will instead use pre-built components [todo: link to this]

## Prepare your build environment

Typically, this process only needs to be run once, although periodically you
may need to repeat some steps (eg, rust updates should be done periodically)

### Setting up Android dependencies

At the end of this process you should have the following environment variables set up.

- `ANDROID_HOME`
- `NDK_HOME`

The easiest way to install all the dependencies (and automatically
handle updates), is by using [Android Studio](https://developer.android.com/studio/index.html).
Once this is installed, the following dependencies can be installed through it:

- Android SDK Tools
- NDK
- CMake
- LLDB

To install a dependency, open Android Studio and select `Tools > SDK Manager > SDK Tools`. Then tick the boxes corrensponding to the dependencies listed above.

> With the dependencies installed, note down the `Android SDK Location` in `Tools > SDK Manager`. Set the `ANDROID_HOME` environment variable to that path. The `NDK_HOME` can be set to `NDK_HOME=$ANDROID_HOME/ndk-bundle`.

Reviewer: I didn't need to set `NDK_HOME`. Once I installed the ndk through the sdk manager it picked it up automatically.

Author: Oh, I see. This seems required on my desktop, but not required on my laptop :\ Should this stay or should it go? :D

Reviewer: If it was needed, keep it in. It certainly won't break stuff.

### Setting up Rust

Rust can be installed using `rustup`, with the following commands:

- `curl https://sh.rustup.rs -sSf | sh`
- `rustup update`

> Platform specific toolchains need to be installed in Rust. This can be

Reviewer: ```suggestion
Platform specific toolchains need to be installed for Rust. This can be
```

> done using `rustup target` command. In order to enable building to real

Reviewer: ```suggestion
done using the `rustup target` command. In order to enable building to real
```

devices and Android emulators, the following targets need to be installed:

- `rustup target add aarch64-linux-android`
- `rustup target add armv7-linux-androideabi`
- `rustup target add i686-linux-android`
- `rustup target add x86_64-linux-android`

## Building

This should be relatively straightforward and painless:

> 1. Ensure your clone repo is up-to-date.

Reviewer: ```suggestion
1. Ensure your repository is up-to-date.
```

Let's avoid abbrevations in text

   You can see a list of projects by executing `./gradlew projects` and a list
   of tasks by executing `./gradlew tasks`.

> The above can be skipped if using `Android Studio`, as the project directory can be imported

Reviewer: ```suggestion
The above can be skipped if using `Android Studio`.
The project directory can be imported
```

and all the build details can be left to the IDE.

> ### Publishing the components to your local maven repository

Reviewer: While we're at it: should we add a note that this is incomplete? (We didn't setup/test publishing to maven)

Author: I vote for removing this version, since we didn't test it. We can always look it up later.
